### PR TITLE
New Changes to Domination Round/Turn

### PR DESCRIPTION
**Pull request by ZoeKHarvey**
- Fixed some bugs in our round and turn files
- started working on  Domination round and turn (functionality and tests)

### CODE REVIEW

**Comment by edwindelbosque** [comment]
@brittanystoroz @khalidwilliams 

We are having trouble whether the inherited fast turn and fast round will have all of the methods invoked by the normal round and turn.

For DominationTurn (fast turn) we need a score multiplier, however,  we see that in our regular turn we have a method that updates the score of the player based on the points of the answer. we DON'T want that. We want to be able to keep that update score method to stay/execute only in the normal round, and the multiplier score to run only in DominationTurn.

We were thinking if perhaps it was the most appropriate idea to have a very generic class template for turn where then we can have two inherited classes (normal turn = normal score update) and (domination turn = multiplied score update)

**Comment by khalidwilliams** [comment]
You could extend two kinds of Rounds from your base class, or I believe you could overwrite the behavior in your extended class (ie write the scoring method in the fastRound class, but name it the same thing). The former is probably better practice, but both would work

File: src/Turn.js [hunk 1]
```diff
@@ -54,31 +55,11 @@ class Turn {
     return this.checkGuess(guess) ? 'Correct!' : 'Incorrect!'
   }
 
-  toggleHelper() {
+  togglePlayer() {
```
Comment (khalidwilliams) [inline src/Turn.js]: better method name, nice job!

File: test/fast-money-test.js [hunk 2]
```diff
@@ -1,22 +1,32 @@
...
-//     expect(DominationRound).to.be.a('function');
-//   });
+  it.only('should be a function', () => {
+    expect(DominationRound).to.be.a('function');
+  });
 
-//   it('should be an instance of DominationRound', () => {
-//     expect(dominationRound).to.be.an.instanceOf(DominationRound);
-//   });
+  it.only('should be an instance of DominationRound', () => {
```
Comment (khalidwilliams) [inline test/fast-money-test.js]: Be careful with the `.only` chain, you don't want to forget to run all your tests from time to time

File: test/fast-money-test.js [hunk 3]
```diff
@@ -1,22 +1,32 @@
...
+  //   dominationRound.startDominationRound()
 
-// }); // <------ end of describe block
\ No newline at end of file
+  // })
+
+  
+
+
+}); // <------ end of describe block
```
Comment (khalidwilliams) [inline test/fast-money-test.js]: Take out this comment, you shouldn't have to label where your blocks end (indentation should speak for itself)

File: test/fast-turn-test.js [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+
+  it.only('should multiply scores based on users multiplier', () => {
+    expect(dominationTurn.calculateScores())
+  })
+
+
+  
+
+
+}); // <------ end of describe block
```
Comment (khalidwilliams) [inline test/fast-turn-test.js]: See previous comments in `test/fast-money-test.js`

File: test/game-test.js [hunk 5]
```diff
@@ -68,6 +68,17 @@ it('should only pick a survey that has not been used', () => {
   { answer: 'Dryer Sheets', respondents: 6, surveyId: 8}])
 })
 
+it('should call something', () => {
+  game.getWinner()
+  expect(DOMupdates.showWinner).to.have.been.called(1)
```
Comment (khalidwilliams) [inline test/game-test.js]: I spy with my little eye, a good example of a chai spy

File: test/game-test.js [hunk 6]
```diff
@@ -68,6 +68,17 @@ it('should only pick a survey that has not been used', () => {
   { answer: 'Dryer Sheets', respondents: 6, surveyId: 8}])
 })
 
+it('should call something', () => {
```
Comment (khalidwilliams) [inline test/game-test.js]: Get more specific with this `it` block title -- what should be called? Remember, the tests should be written clearly enough that they can serve as documentation

File: test/game-test.js [hunk 7]
```diff
@@ -82,6 +93,7 @@ it('should only pick a survey that has not been used', () => {
 
 }); // <------ end of describe block
 
+
 describe('getWinner', function() {
   it('should call something', () => {
```
Comment (khalidwilliams) [inline test/game-test.js]: Get more specific with these it block titles!

File: test/game-test.js [hunk 8]
```diff
@@ -68,6 +68,17 @@ it('should only pick a survey that has not been used', () => {
...
 })
 
+it('should call something', () => {
+  game.getWinner()
+  expect(DOMupdates.showWinner).to.have.been.called(1)
+})
+
+// it('should create a regular round, () => {
+//   game.startGame();
+//   chai.spy.on(game.currentRound, 'beginRound', () => true)
```
Comment (khalidwilliams) [inline test/game-test.js]: Why are you spying here? Is `beginRound` calling domUpdate methods?

File: src/Turn.js [hunk 9]
```diff
@@ -5,15 +5,16 @@ class Turn {
...
   }
 
   identifyQuestion() {
     return this.currentRound.survey[0].id
   }
 
   identifyAnswerInfo() {
-    return this.currentRound.answers;
+    return this.currentRound.survey.filter(survey => {
+      return survey.answer;
```
[the remark this answers is not in the document]
Comment (edwindelbosque) [reply src/Turn.js]: Yeah I found this bug earlier today...  thanks @khalidwilliams

**Pull request merged**